### PR TITLE
Make the internal http client return the raw TaskEither and not the Promise

### PR DESCRIPTION
**Pull request by XVincentX**
This PR modifies the internal client we're using in the http server and in some other places to no throw exceptions anymore.

Since this is an _internal_ client, nobody should use it and so nobody will get hurt. The *external* client stays the same.

I've made all the necessary changes that might make the code uglier at this point, but the idea would be that we need to ditch fastify since:

1. We're using 1% of its features
2. Some of the TypeScript definitions are wrong
3. It's not friendly to the programming style we're using.

Most likely, we're going to use `micri` which proved to be small and barebone/close to the metal enough. In meantime, we'll live with this.

### CODE REVIEW

File: packages/core/src/types.ts [hunk 1]
```diff
@@ -45,8 +45,7 @@ export interface IPrismInput<I> {
   validations: IPrismDiagnostic[];
 }
 
-export interface IPrismOutput<I, O> {
-  input: I;
+export interface IPrismOutput<O> {
```
Comment (StefanDywersant) [inline packages/core/src/types.ts]: Note to self: why?

File: packages/http/README.md [hunk 2]
```diff
@@ -71,20 +71,18 @@ const prism = Prism.createInstance({
...
-  .process(
-    {
-      method: 'get',
-      url: {
-        path: '/todos',
-      },
-      headers: {
-        Accept: 'text/plain',
-      },
+return prism.process(
```
Comment (StefanDywersant) [inline packages/http/README.md]:
```suggestion
return prism.request(
```

File: packages/http/README.md [hunk 3]
```diff
@@ -71,20 +71,18 @@ const prism = Prism.createInstance({
...
     },
-    operations
-  )
-  .then(prismResponse => console.log(prismResponse.output));
+    headers: {
+      Accept: 'text/plain',
+    },
+  },
+  operations
+);
```
Comment (StefanDywersant) [inline packages/http/README.md]:
```suggestion
)();
```
+ `TaskEither.fold` should be there?

File: packages/http/README.md [hunk 4]
```diff
@@ -110,24 +108,22 @@ const Prism = require('@stoplight/prism-http');
...
 // Make a "GET /todos" request
-return prism
-  .process(
-    {
-      method: 'get',
-      url: {
-        path: '/facts',
-        baseUrl: 'https://cat-fact.herokuapp.com',
-      },
+return prism.process(
```
Comment (StefanDywersant) [inline packages/http/README.md]:
```suggestion
return prism.request(
```

File: packages/http/README.md [hunk 5]
```diff
@@ -110,24 +108,22 @@ const Prism = require('@stoplight/prism-http');
...
+    // We can override the default behaviour per request.
+    mock: {
+      dynamic: true,
     },
-    operations
-  )
-  .then(prismResponse => console.log(prismResponse.output));
+  },
+  operations
+);
```
Comment (StefanDywersant) [inline packages/http/README.md]:
```suggestion
)();
```
.. and `TaskEither.fold` should be there?

File: packages/http/README.md [hunk 6]
```diff
@@ -316,18 +310,16 @@ const Prism = require('@stoplight/prism-http');
...
+    url: {
+      path: '/facts',
+      baseUrl: 'https://cat-fact.herokuapp.com',
     },
-    operations
-  )
-  .then(prismResponse => console.log(prismResponse.validations.input));
+  },
+  operations
+);
```
Comment (StefanDywersant) [inline packages/http/README.md]: ditto

File: packages/http/README.md [hunk 7]
```diff
@@ -316,18 +310,16 @@ const Prism = require('@stoplight/prism-http');
...
 const prism = Prism.createInstance(config);
-prism
-  .process(
-    {
-      method: 'get',
-      url: {
-        path: '/facts',
-        baseUrl: 'https://cat-fact.herokuapp.com',
-      },
+prism.process(
```
Comment (StefanDywersant) [inline packages/http/README.md]:
```suggestion
prism.request(
```

File: packages/http/README.md [hunk 8]
```diff
@@ -141,15 +137,13 @@ const Prism = require('@stoplight/prism-http');
...
-    },
-  })
-  .then(prismResponse => console.log(prismResponse.output));
+return prism.request({
+  method: 'get',
+  url: {
+    path: '/facts',
+    baseUrl: 'https://cat-fact.herokuapp.com',
+  },
+});
```
Comment (StefanDywersant) [inline packages/http/README.md]: ditto

File: packages/http-server/src/server.ts [hunk 9]
```diff
@@ -60,72 +63,78 @@ export const createServer = (operations: IHttpOperation[], opts: IPrismHttpServe
...
-        );
-        if (opts.errors && errorViolations.length > 0) {
-          throw ProblemJsonError.fromTemplate(
-            VIOLATIONS,
-            'Your request/response is not valid and the --errors flag is set, so Prism is generating this error for you.',
-            { validation: errorViolations }
+          const errorViolations = inputOutputValidationErrors.filter(
+            v => v.severity === DiagnosticSeverity[DiagnosticSeverity.Error]
           );
+          if (opts.errors && errorViolations.length > 0) {
```
Comment (StefanDywersant) [inline packages/http-server/src/server.ts]: I believe this code still needs some love and we will do it later?

File: packages/http/src/__tests__/http-prism-instance.spec.ts [hunk 10]
```diff
@@ -65,43 +66,49 @@ describe('Http Client .request', () => {
...
 
     describe('baseUrl not set', () => {
-      it('ignores server validation and returns 200', async () => {
-        const result = await prism.request(
-          {
-            method: 'get',
-            url: {
-              path: '/pet',
+      it('ignores server validation and returns 200', () => {
+        assertResolvesRight(
```
Comment (StefanDywersant) [inline packages/http/src/__tests__/http-prism-instance.spec.ts]: Hello, my lovely helper function :)

File: packages/core/src/types.ts (hunk 1 above)
Comment (StefanDywersant) [reply packages/core/src/types.ts]: I guess `Input` wasn't used, right?

Comment (XVincentX) [reply packages/core/src/types.ts]: Correct, it was not used at all

File: packages/http-server/src/server.ts (hunk 9 above)
Comment (XVincentX) [reply packages/http-server/src/server.ts]: Yes, this was the minimal part to move it over. We'll keep iterating on this.

File: packages/http/src/__tests__/http-prism-instance.spec.ts (hunk 10 above)
Comment (XVincentX) [reply packages/http/src/__tests__/http-prism-instance.spec.ts]: That was a good one! 👍

File: packages/http/README.md (hunk 5 above)
Comment (XVincentX) [reply packages/http/README.md]: Yes, I'll update the docs.

**Comment by XVincentX** [comment]
> Code looks good, except the big nest of if's in server.ts, but I guess that is the part that will be reworked later.

Yes, we also need to dismantle this part of the application and make it easier.

> In order to this comment be more constructive: Maybe we should introduce two interfaces. Classic and FP?

This is happening already. We basically have two clients: the internal one (which is the one being modified by this Pull Request) and the public one (which is in the `client.ts` file); this one has not been modified at all. Does this clarify?

File: packages/http/README.md [hunk 11]
```diff
@@ -97,6 +95,26 @@ Output
...
+        url: {
+          path: '/todos',
+        },
+        headers: {
+          Accept: 'text/plain',
+        },
+      },
+      operations
+    ),
+    TaskEither.fold(console.error, data => console.log(data));
```
Comment (StefanDywersant) [inline packages/http/README.md]:
```suggestion
    TaskEither.fold(console.error, console.log);
```

**Comment by XVincentX** [comment]
@StefanDywersant The client has a dedicated help page.

**Pull request merged**